[DC-1663] Fix bad test smells

lrwb-aou opened the pull request:
* removes unnecessary table drops and recreations.  all tables are created with defined schemas by base class.
* utilizes `cls.fq_table_names` as intended to create/delete tables from base class.
* removes superfluous copy function in favor of existing client object copy function.
* removes creating new client objects in tests when the base client object is already available.
* removes redundant test teardowns that are overwriting the same definition in the base class.

dcampbell-vumc reviewed (changes requested):
I ran every integration test to ensure everything passed as normal. Searched the project for `DROP TABLE IF EXISTS` and didn't find any references in any of the integration tests. 

There are some comments that could be confusing because it references needing to drop the table even though we have removed that functionality so those can be removed right?

lrwb-aou commented on tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/clean_height_weight_test.py:
```diff
@@ -81,32 +81,27 @@
 # partitioning by _PARTITIONTIME doesn't work using a query_statement for creating a table,
 # therefore CREATE OR REPLACE TABLE doesn' work and we need to DROP the table first.
```
correct.  nice catch.

lrwb-aou commented on tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/no_data_30_days_after_death_test.py:
```diff
@@ -24,57 +24,41 @@
 # partitioning by _PARTITIONTIME doesn't work using a query_statement for creating a table,
 # therefore CREATE OR REPLACE TABLE doesn' work and we need to DROP the table first.
```
good catch.

lrwb-aou commented:
> I ran every integration test to ensure everything passed as normal. Searched the project for `DROP TABLE IF EXISTS` and didn't find any references in any of the integration tests.
> 
> There are some comments that could be confusing because it references needing to drop the table even though we have removed that functionality so those can be removed right?

you are right.  i've removed those comments and pushed those changes.

dcampbell-vumc reviewed (approved):
This is good to go once the integration check passes :D
